diff --git a/genode/repos/gems/src/app/themed_decorator/theme.cc b/genode/repos/gems/src/app/themed_decorator/theme.cc
index a8882e1516..1d2f101c84 100644
--- a/genode/repos/gems/src/app/themed_decorator/theme.cc
+++ b/genode/repos/gems/src/app/themed_decorator/theme.cc
@@ -27,7 +27,10 @@
 #include "theme.h"
 
 
-enum Texture_id { TEXTURE_ID_DEFAULT, TEXTURE_ID_CLOSER, TEXTURE_ID_MAXIMIZER };
+enum Texture_id { TEXTURE_ID_DEFAULT, TEXTURE_ID_CLOSER, TEXTURE_ID_MAXIMIZER,
+                       TEXTURE_ID_SIZER_NW, TEXTURE_ID_SIZER_NE,
+                       TEXTURE_ID_SIZER_SW, TEXTURE_ID_SIZER_SE,
+                       TEXTURE_ID_MINIMIZER };
 
 
 struct Texture_from_png_file
@@ -65,6 +68,31 @@ texture_by_id(Genode::Ram_allocator &ram, Genode::Env::Local_rm &rm,
 		return texture.texture;
 	}
 
+	if (texture_id == TEXTURE_ID_SIZER_NW) {
+		static Texture_from_png_file texture(ram, rm, alloc, "theme/sizer_nw.png");
+		return texture.texture;
+	}
+
+	if (texture_id == TEXTURE_ID_SIZER_NE) {
+		static Texture_from_png_file texture(ram, rm, alloc, "theme/sizer_ne.png");
+		return texture.texture;
+	}
+
+	if (texture_id == TEXTURE_ID_SIZER_SW) {
+		static Texture_from_png_file texture(ram, rm, alloc, "theme/sizer_sw.png");
+		return texture.texture;
+	}
+
+	if (texture_id == TEXTURE_ID_SIZER_SE) {
+		static Texture_from_png_file texture(ram, rm, alloc, "theme/sizer_se.png");
+		return texture.texture;
+	}
+
+	if (texture_id == TEXTURE_ID_MINIMIZER) {
+		static Texture_from_png_file texture(ram, rm, alloc, "theme/minimizer.png");
+		return texture.texture;
+	}
+
 	struct Invalid_texture_id { };
 	throw  Invalid_texture_id();
 }
@@ -80,6 +108,21 @@ texture_by_element_type(Genode::Ram_allocator &ram, Genode::Env::Local_rm &rm,
 
 	case Decorator::Theme::ELEMENT_TYPE_MAXIMIZER:
 		return texture_by_id(ram, rm, alloc, TEXTURE_ID_MAXIMIZER);
+
+	case Decorator::Theme::ELEMENT_TYPE_SIZER_NW:
+		return texture_by_id(ram, rm, alloc, TEXTURE_ID_SIZER_NW);
+
+	case Decorator::Theme::ELEMENT_TYPE_SIZER_NE:
+		return texture_by_id(ram, rm, alloc, TEXTURE_ID_SIZER_NE);
+
+	case Decorator::Theme::ELEMENT_TYPE_SIZER_SW:
+		return texture_by_id(ram, rm, alloc, TEXTURE_ID_SIZER_SW);
+
+	case Decorator::Theme::ELEMENT_TYPE_SIZER_SE:
+		return texture_by_id(ram, rm, alloc, TEXTURE_ID_SIZER_SE);
+
+	case Decorator::Theme::ELEMENT_TYPE_MINIMIZER:
+		return texture_by_id(ram, rm, alloc, TEXTURE_ID_MINIMIZER);
 	}
 	struct Invalid_element_type { };
 	throw  Invalid_element_type();
@@ -184,6 +227,21 @@ Decorator::Rect Decorator::Theme::element_geometry(Element_type type) const
 	if (type == ELEMENT_TYPE_MAXIMIZER)
 		return ::element_geometry(_ram, _rm, _alloc, "maximizer", TEXTURE_ID_MAXIMIZER);
 
+	if (type == ELEMENT_TYPE_SIZER_NW)
+		return ::element_geometry(_ram, _rm, _alloc, "sizer_nw", TEXTURE_ID_SIZER_NW);
+
+	if (type == ELEMENT_TYPE_SIZER_NE)
+		return ::element_geometry(_ram, _rm, _alloc, "sizer_ne", TEXTURE_ID_SIZER_NE);
+
+	if (type == ELEMENT_TYPE_SIZER_SW)
+		return ::element_geometry(_ram, _rm, _alloc, "sizer_sw", TEXTURE_ID_SIZER_SW);
+
+	if (type == ELEMENT_TYPE_SIZER_SE)
+		return ::element_geometry(_ram, _rm, _alloc, "sizer_se", TEXTURE_ID_SIZER_SE);
+
+	if (type == ELEMENT_TYPE_MINIMIZER)
+		return ::element_geometry(_ram, _rm, _alloc, "minimizer", TEXTURE_ID_MINIMIZER);
+
 	struct Invalid_element_type { };
 	throw  Invalid_element_type();
 }
diff --git a/genode/repos/gems/src/app/themed_decorator/theme.h b/genode/repos/gems/src/app/themed_decorator/theme.h
index 9a1bc73fe5..a172b99d80 100644
--- a/genode/repos/gems/src/app/themed_decorator/theme.h
+++ b/genode/repos/gems/src/app/themed_decorator/theme.h
@@ -50,7 +50,10 @@ class Decorator::Theme
 			bool none() const { return !top && !bottom && !left && !right; }
 		};
 
-		enum Element_type { ELEMENT_TYPE_CLOSER, ELEMENT_TYPE_MAXIMIZER };
+		enum Element_type { ELEMENT_TYPE_CLOSER, ELEMENT_TYPE_MAXIMIZER,
+		                    ELEMENT_TYPE_SIZER_NW, ELEMENT_TYPE_SIZER_NE,
+		                    ELEMENT_TYPE_SIZER_SW, ELEMENT_TYPE_SIZER_SE,
+		                    ELEMENT_TYPE_MINIMIZER };
 
 		Theme(Genode::Ram_allocator &ram, Genode::Env::Local_rm &rm,
 		      Genode::Allocator &alloc)
diff --git a/genode/repos/gems/src/app/themed_decorator/window.cc b/genode/repos/gems/src/app/themed_decorator/window.cc
index 5af4f7147a..69aa82102a 100644
--- a/genode/repos/gems/src/app/themed_decorator/window.cc
+++ b/genode/repos/gems/src/app/themed_decorator/window.cc
@@ -43,6 +43,14 @@ Decorator::Window_base::Hover Decorator::Window::hover(Point abs_pos) const
 		return hover;
 	}
 
+	Rect const minimizer_geometry =
+		_theme.absolute(_theme.element_geometry(Theme::ELEMENT_TYPE_MINIMIZER),
+		                                        outer_geometry());
+	if (_minimizer.present() && minimizer_geometry.contains(abs_pos)) {
+		hover.minimizer = true;
+		return hover;
+	}
+
 	Rect const title_geometry = _theme.absolute(_theme.title_geometry(),
 	                                            outer_geometry());
 	if (title_geometry.contains(abs_pos)) {
diff --git a/genode/repos/gems/src/app/themed_decorator/window.h b/genode/repos/gems/src/app/themed_decorator/window.h
index 452d2c102c..fba5857a3a 100644
--- a/genode/repos/gems/src/app/themed_decorator/window.h
+++ b/genode/repos/gems/src/app/themed_decorator/window.h
@@ -135,12 +135,14 @@ class Decorator::Window : public Window_base, public Animator::Item
 
 		Element _closer    { _animator, Theme::ELEMENT_TYPE_CLOSER,    "closer" };
 		Element _maximizer { _animator, Theme::ELEMENT_TYPE_MAXIMIZER, "maximizer" };
+		Element _minimizer { _animator, Theme::ELEMENT_TYPE_MINIMIZER, "minimizer" };
 
 		template <typename FN>
 		void _for_each_element(FN const &func)
 		{
 			func(_closer);
 			func(_maximizer);
+			func(_minimizer);
 		}
 
 		struct Gui_view : Genode::Noncopyable
@@ -590,6 +592,7 @@ class Decorator::Window : public Window_base, public Animator::Item
 			return (_alpha.dst() != (int)_alpha)
 			    || _r != _r.dst() || _g != _g.dst() || _b != _b.dst()
 			    || _closer.animated() || _maximizer.animated()
+			    || _minimizer.animated()
 			    || _animated_rect.animated();
 		}
 
