Assertion triggered on Windows 10 with cb=0x1000 (investigate?)

diff --git src/virtualbox7/src/VBox/Devices/Bus/DevPciIch9.cpp src/virtualbox7/src/VBox/Devices/Bus/DevPciIch9.cpp
index 6f469ce3..8b4bb2af 100644
--- src/virtualbox7/src/VBox/Devices/Bus/DevPciIch9.cpp
+++ src/virtualbox7/src/VBox/Devices/Bus/DevPciIch9.cpp
@@ -729,7 +729,7 @@ DECL_HIDDEN_CALLBACK(VBOXSTRICTRC) devpciCommonMcfgMmioRead(PPDMDEVINS pDevIns,
                 *(uint32_t *)pv  = u32Value;
                 break;
             default:
-                ASSERT_GUEST_MSG_FAILED(("cb=%u off=%RGp\n", cb, off)); /** @todo how the heck should this work? Split it, right? */
+                //ASSERT_GUEST_MSG_FAILED(("cb=%u off=%RGp\n", cb, off)); /** @todo how the heck should this work? Split it, right? */
                 rcStrict = VINF_IOM_MMIO_UNUSED_00;
                 break;
         }
